Translate CVE-2021-33621 HTTP response splitting in CGI news post (pt) (#2974)
ruby/www.ruby-lang.org: pt/news/_posts/2022-11-22-http-response-splitting-in-cgi-cve-2021-33621.md

---
layout: news_post
title: "CVE-2021-33621: HTTP response splitting na CGI"
author: "mame"
translator: "guicruzzs"
date: 2022-11-22 02:00:00 +0000
tags: security
lang: pt
---

> Nós lançamos gem cgi versão 0.3.5, 0.2.2, e 0.1.0.2 que possui uma correção de segurança para uma vulnerabilidade de HTTP response splitting.

Reviewer: ```suggestion
Nós lançamos as versões da gem cgi 0.3.5, 0.2.2, e 0.1.0.2 que possuem uma correção de segurança para uma vulnerabilidade de HTTP response splitting.
```

> Essa vulnerabilidade foi atribuída ao identificador CVE [CVE-2021-33621](https://nvd.nist.gov/vuln/detail/CVE-2021-33621).

Reviewer: ```suggestion
Essa vulnerabilidade foi atribuída ao identificador [CVE-2021-33621](https://nvd.nist.gov/vuln/detail/CVE-2021-33621).
```

## Detalhes

> Se uma aplicação que gera respostas HTTP usando a gem cgi com uma input não confiável do usuário, um atacante pode explorá-la injetando um body e/ou header malicioso na resposta HTTP.

Reviewer: Eu acho que "com uma input" não ficou tão claro.
Talvez usar "com uma entrada de dados não confiável"?

Author: eu fiquei nessa dúvida também
acho que é uma ótima sugestão

> Também, os conteúdos de um objeto `CGI::Cookie` não eram verificado apropriadamente. Se uma aplicação cria um objeto `CGI::Cookie` baseado no input do usuário, um atacante pode explorá-lo para injetar atributos inválidos no cabeçalho `Set-Cookie`. Nós achamos tais aplicações indesejadas, mas incluímos uma alteração para verificar os argumentos em `CGI::Cookie#initialize` preventivamente.

Reviewer: ```suggestion
Também, os conteúdos de um objeto `CGI::Cookie` não eram verificadow apropriadamente. Se uma aplicação cria um objeto `CGI::Cookie` baseado no input do usuário, um atacante pode explorá-lo para injetar atributos inválidos no cabeçalho `Set-Cookie`. Nós achamos tais aplicações indesejadas, mas incluímos uma alteração para verificar os argumentos em `CGI::Cookie#initialize` preventivamente.
```

Author: eu entendi o plural aqui :+1:

> Por favor atualize a gem cgi para a versão 0.3.5, 0.2.2, e 0.1.0.2, ou maiores. Você pode usar `gem update cgi` para atualizá-la.

Reviewer: ```suggestion
Por favor atualize a gem cgi para as versões 0.3.5, 0.2.2, e 0.1.0.2, ou maiores. Você pode usar `gem update cgi` para atualizá-la.
```
